.ci/update-tekton-bundle.sh

```diff
@@ -0,0 +1,79 @@
+APPSTUDIO_TASKS_REPO=quay.io/repository/redhat-appstudio-qe/test-images
+PIPELINE_BUNDLE_IMG=quay.io/repository/redhat-appstudio-qe/test-images:pipeline-bundle-$JVM_BUILD_SERVICE_PR_SHA
+BUILD_DEFINITIONS_DIR="$WORKSPACE"/tmp/build-definitions
+
+function cloneBuildDefinitions() {
+    git clone https://$GITHUB_TOKEN@github.com/redhat-appstudio/build-definitions.git "$BUILD_DEFINITIONS_DIR"
+}
+
+function updateAnalyzerImage() {
+    yq e -i "(.[] \
+    | select(.path == \"/spec/steps/-\").value \
+    | select(.name == \"analyse-dependencies-java-sbom\").image) \
+    |= \"$JVM_BUILD_SERVICE_ANALYZER_IMAGE\"" \
+    $BUILD_DEFINITIONS_DIR/tasks/patches/add-sbom-and-push.yaml
+}
+
+function createAndPushTaskBundle() {
+    if [[ $(uname) = Darwin ]]; then
+        CSPLIT_CMD="gcsplit"
+    else
+        CSPLIT_CMD="csplit"
+    fi
+
+    PART=1
+    COUNT=0
+    TASK_TEMP=$(mktemp -d)
+    PIPELINE_TEMP=$(mktemp -d)
+    oc kustomize $BUILD_DEFINITIONS_DIR/tasks | $CSPLIT_CMD -s -f $TASK_TEMP/task -b %02d.yaml - /^---$/ '{*}'
+    oc kustomize $BUILD_DEFINITIONS_DIR/pipelines/base > ${PIPELINE_TEMP}/base.yaml
+
+    ## Limit number of tasks in bundle
+    MAX=10
+    TASKS=
+    REF="$APPSTUDIO_TASKS_REPO:task-bundle-$JVM_BUILD_SERVICE_PR_SHA-$PART"
+    for TASK in $TASK_TEMP/task*.yaml; do
+        TASK_NAME=$(yq e ".metadata.name" $TASK)
+        if [ $COUNT -eq $MAX ]; then
+            echo Creating $REF
+            tkn bundle push $TASKS $REF
+            COUNT=0
+            PART=$((PART+1))
+            TASKS=
+        fi
+        REF="$APPSTUDIO_TASKS_REPO:task-bundle-$JVM_BUILD_SERVICE_PR_SHA-$PART"
+
+        # Replace appstudio-utils placeholder by newly build appstudio-utils image
+        #yq e -i ".spec.steps[] |= select(.image == \"appstudio-utils\").image=\"$APPSTUDIO_UTILS_IMG\"" $TASK
+
+        TASKS="$TASKS -f $TASK"
+        for file in $PIPELINE_TEMP/*.yaml; do
+            yq e -i "(.spec.tasks[].taskRef | select(.name == \"$TASK_NAME\")) |= {\"name\": \"$TASK_NAME\", \"bundle\":\"$REF\"}" $file
+            yq e -i "(.spec.finally[].taskRef | select(.name == \"$TASK_NAME\")) |= {\"name\": \"$TASK_NAME\", \"bundle\":\"$REF\"}" $file
+        done
+        COUNT=$((COUNT+1))
+    done
+    echo Creating $APPSTUDIO_TASKS_REPO:task-bundle-$JVM_BUILD_SERVICE_PR_SHA-$PART
+    tkn bundle push $TASKS $REF
+}
+
+function createAndPushPipelineBundles() {
+    tkn bundle push $PIPELINE_BUNDLE_IMG -f ${PIPELINE_TEMP}/base.yaml
+}
+
+function createConfigMap() {
+    oc apply -m build-templates -f - << CONFIGMAP
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: build-pipelines-defaults
+data:
+  default_build_bundle: "$PIPELINE_BUNDLE_IMG"
```

Comment:
so this overrides the pipeline bundle

don't we need a similar entry for the task bundle?

Reply:
Won't the rebuilt build bundle reference the rebuilt task bundles that were created in the build process?

Reply:
maybe?

can you post some sample output that clarifies one way or the other @brunoapimentel ?  or maybe add a "debug commit" that runs `cat` on the relevant updated yaml prior to calling `tkn` ?

Reply:
In this script, we're updating all the task refs to point to the newly built ones: https://github.com/brunoapimentel/jvm-build-service/blob/ci-override-analyzer-image/.ci/update-tekton-bundle.sh#L49-L53

Reply:
ok that helps @brunoapimentel thanks

just to make sure I'm understanding the bundle structure, is it correct to say that while we have 3 bundles, the pipeline bundles is the "top level bundle" with refs to the task bundle and conceivably the util bundle as well?